### PR TITLE
feat(knowledge): ADR-0010 index, entries output, validate freshness

### LINKED ISSUE #52: feat(knowledge): standardize knowledge format and manifest generation
## Context
Implement ADR-0010: repository knowledge format, `rgd knowledge index`, manifest generation from YAML front matter, `knowledge.entries` in operational context, optional `rgd knowledge pack --query`, and extended `rgd config validate` (paths, freshness, hints).

## Refs: ADR
- ADR-0010 (`docs/adr/0010-knowledge-management.md`)
- ADR-0008 (schema versioning)

## ADR Impact
none (ADR merged in same PR or preceding commit)

## Definition of Done
- [ ] ADR-0010 present
- [ ] Schemas + Go types + `internal/knowledge/`
- [ ] CLI: `knowledge index`, `knowledge pack` (`--query`, `entries` output), `context build` entries, `config validate` extensions
- [ ] `.reinguard/knowledge` migrated; redundant files removed
- [ ] `docs/cli.md` + Cursor bridge rule updated
- [ ] Tests + `go vet` + `golangci-lint`

## Test plan
| Case | Expected |
|------|----------|
| `rgd knowledge index` | Writes manifest from front matter |
| `rgd config validate` stale manifest | Non-zero |
| `rgd knowledge pack --query x` | Filtered entries |
| Golden `context build` | `knowledge.entries` shape |

### PR DESCRIPTION
**Pull request by k-shibuki**
## Summary

- Add ADR-0010 and implement `rgd knowledge index`, manifest generation from Markdown front matter, and extended `rgd config validate` (paths, freshness, hints).
- Operational context and `rgd knowledge pack` now expose `knowledge.entries` (id, path, description, triggers); optional `--query` filters by trigger substring.
- Migrate `.reinguard/knowledge` to the new format; add `.cursor/rules/knowledge-bridge.mdc`; bump synchronized `schema_version` to 0.3.0.

## Traceability

Closes #52

Refs: #52

## Definition of Done

- [x] Tests added or updated (`go test ./...`)
- [x] `go vet ./...` clean
- [x] Lint clean (golangci-lint / CI)
- [x] Documentation updated (ADR-0010, `docs/cli.md`, `docs/design.md`, `AGENTS.md`)

## Test plan

1. `go test ./...` — all packages pass.
2. `go vet ./...` and `golangci-lint run` — clean.
3. `go run ./cmd/rgd knowledge index` — regenerates `.reinguard/knowledge/manifest.json`.
4. `go run ./cmd/rgd config validate` — succeeds with freshness and path checks.
5. `go run ./cmd/rgd knowledge pack` / `context build` — JSON includes `entries` array as documented.

## Risk / Impact

Breaking change for any consumer of operational context `knowledge.paths` (removed; use `knowledge.entries`). `knowledge/manifest.json` schema requires `description` and `triggers` per entry; repos must run `rgd knowledge index` after editing knowledge front matter. CI `config validate` will fail if manifest is stale.

## Rollback Plan

Revert the merge commit on `main` (or `git revert <merge-sha>`). Downstream repos that adopted 0.3.0 manifests may need to restore prior manifest shape or pin an older `rgd` release until they migrate.

### CODE REVIEW

**Comment by k-shibuki** [comment]
@coderabbitai review

File: .reinguard/knowledge/review--source-summary.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+Most-commented paths by CodeRabbit root comments:
+1. `internal/rgdcli/rgdcli.go` (7)
+2. `.cursor/rules/workflow-policy.mdc` (5)
+3. `internal/observe/provider_git.go` (4)
+4. `internal/rgdcli/flatten_test.go` (4)
+5. `tools/backfill-open-pr-policy.py` (4)
+
+Recurring finding clusters (keyword-derived):
+- JSON contract / shape assertions: high frequency
+- Test rigor (Given/When/Then, boundary cases, explicit outcomes): high frequency
```
[the remark this answers is not in the document]
Comment (k-shibuki) [reply .reinguard/knowledge/review--source-summary.md]: **Fixed.** Lines 35–36 now use hyphenated “high-frequency” in `review--source-summary.md` (93f9f1fbf718f15a955be41ed484582425578fe8).
